### ISSUE TITLE
Segfault when creating video mode 

### ISSUE THREAD

**Issue by Masy** [glfw/glfw#1950]
I had a break from a project of mine for a little over a month and suddenly it stopped working because of a segfault when
initializing my window wrapper class.

This is the method that is giving me a segfault when calling glfwGetVideMode:
```cpp
int PWindow::init(const int selectedMonitor, const bool centerOnMonitor)
{
	int monitorCount = 0;
	GLFWmonitor **monitors = glfwGetMonitors(&monitorCount);

	if (monitorCount == 0 || monitors == nullptr)
		throw PException("Could not find any monitors!");

	if (selectedMonitor >= monitorCount)
		this->glfwMonitor = monitors[0];
	else
		this->glfwMonitor = monitors[selectedMonitor];

	this->glfwVidmode = glfwGetVideoMode(this->glfwMonitor);

	[...]
}
```

This is the stacktrace:
```
_glfwPlatformGetVideoMode x11_monitor.c:498
glfwGetVideoMode monitor.c:449
PWindow::init PWindow.cpp:355
PWindow::PWindow PWindow.cpp:25
MainWindow::MainWindow MainWindow.cpp:9
main main.cpp:10
__libc_start_main 0x00007fd3303460b3
_start 0x000056303baf444e
```

Which segfaults at ``XRRCrtcInfo* ci = XRRGetCrtcInfo(_glfw.x11.display, sr, monitor->x11.crtc);`` (i pulled before creating the issue, but just in case). I'm on Ubuntu 20.04 with GCC 9.3.0 if that's relevant.

**Comment by elmindreda** [comment]
Your code snipped really should work. Have you tried running the `monitors` test program that comes with GLFW? That could be useful for reproducing the bug.

Are you running in a debugger? It would be helpul to know whether it is `sr` or `monitor->x11.crtc` that is (I assume) NULL.

**Comment by Masy** [comment]
I found the problem: Apparently I edited the selectedMonitor parameter to be an int instead of size_t (unsigned long long for me).
Since i pass "-1" for the primary monitor, i must have missed that before it would fall under the ``selectedMonitor > monitorCount``
branch. Kinda surprised though that i can access an array with -1 and still get a monitor back. (CLion showed a non-null address, although i would assume they were invalid addresses, hence the segfault down the line).
My bad.
